### PR TITLE
fix iter may not be Released in some cases

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
When error happened in loop,  iter may not be released. And when `l0Num >= l0Trigger`, we better release the iter, it may be affect compaction operation.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

**Comment by leoppro** [comment]
@suzaku PTAL

File: pump/storage/storage.go [hunk 1]
```diff
@@ -681,13 +688,24 @@ func (a *Append) doGCTS(ts int64) {
...
+			}
 			time.Sleep(5 * time.Second)
 			continue
 		}
 
 		deleteBatch := 0
 		var lastKey []byte
 
+		if iter == nil {
+			log.Info("create new gc iter", zap.Int64("ts", ts),
```
Comment (suzaku) [inline pump/storage/storage.go]:
```suggestion
			log.Info("New LevelDB iterator created for GC", zap.Int64("ts", ts),
```

**Pull request merged**